### PR TITLE
feat: support for subnets

### PR DESCRIPTION
**Pull request by zone117x**
Closes https://github.com/hirosystems/stacks-blockchain-api/issues/1237

This PR takes over https://github.com/hirosystems/stacks-blockchain-api/pull/1340

The easiest way to review this PR is to look over the integration tests here: https://github.com/hirosystems/stacks-blockchain-api/blob/feat/subnets/src/tests-subnets/subnet-tests.ts

Tests:
* [x] Integration tests for subnets running in CI
* [x] e2e test for NFT-use-case
* [x] e2e test for FT-use-case
  * _Note: subnets has not yet implemented support for an endpoint required to test `withdraw-ft-asset`_
* [x] e2e test for STX-use-case

Implementation:
* [x] Support parsing L1 NFT deposits into synthetic txs
* [x] Support parsing L1 FT deposits into synthetic txs
* [x] Support parsing L1 STX deposits into synthetic txs

Note: this was developed against subnet branch: https://github.com/hirosystems/stacks-subnets/pull/206

### CODE REVIEW

File: src/event-stream/reader.ts [hunk 1]
```diff
@@ -373,6 +544,21 @@ export function parseMessageTransaction(
...
+      const nftMintEvent = events.find(
+        (e): e is NftMintEvent => e.type === CoreNodeEventType.NftMintEvent
+      );
+
+      // {"committed":true,"event_index":0,"ft_mint_event":{"amount":"1","asset_identifier":"ST11NJTTKGVT6D1HY4NJRVQWMQM7TVAR091EJ8P2Y.simple-ft-l2::ft-token","recipient":"ST11NJTTKGVT6D1HY4NJRVQWMQM7TVAR091EJ8P2Y"},"txid":"0xb942a6452312c8388f775d84e80213e6db4e53214f700e3537ecb3d282ab2c79","type":"ft_mint_event"}
+      const ftMintEvent = events.find(
+        (e): e is FtMintEvent => e.type === CoreNodeEventType.FtMintEvent
+      );
+
+      // {"committed":true,"event_index":0,"stx_mint_event":{"amount":"12345678","recipient":"ST1HB1T8WRNBYB0Y3T7WXZS38NKKPTBR3EG9EPJKR"},"txid":"0xde4c90ed393e12e243fcf7573a214dc57b394d273659d3c5d8a8d5868b272bde","type":"stx_mint_event"}
```
Comment (rafaelcr) [inline src/event-stream/reader.ts:557]: These comments seem too specific, were they debug comments? Can we remove them?

File: src/tests-subnets/global-setup.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  eventServer: EventStreamServer;
+}
+
+async function standByForInfoToBeReady(client: StacksCoreRpcClient): Promise<void> {
+  let tries = 0;
+  while (true) {
+    try {
+      tries++;
+      const info = await client.getInfo();
+      // const account = await client.getAccount('ST12KYHMD4GJZJHAPM4VC0VZTB4FEWTQSY74YHEYN');
```
Comment (rafaelcr) [inline src/tests-subnets/global-setup.ts:21]: This also looks like a debug comment

**Review by rafaelcr — APPROVED** [review]
LGTM 💯 just a few nits

**Review by obycode — APPROVED** [review]
Looks good to me too!

**Pull request merged**